src/ucs/async/async.h

```diff
@@ -118,6 +119,24 @@ static inline int ucs_async_is_blocked(const ucs_async_context_t *async)
     return async->poll_block > 0;
 }
 
+/**
+ * Returns wether or not thread ownership has changed
+ * @param async Event context to check for ownership.
+ *
+ * @return 1 only if ownership is unchanged
+ */
+static inline int ucs_async_is_single_thread(ucs_async_context_t *async)
+{
+    if (async->owner != pthread_self()) {
+        if (async->owner != 0) {
+            return 0;
+        }
+
+        async->owner = pthread_self();
```

Comment:
But it also assigns a new value, not just checks
Does it make sense to assign ownership somewhere else, and make async const here?

Reply:
renamed to *_check_*. Intent was to set it on first usage, in case some inits would be done on different thread.